### ISSUE TITLE
A bit more on Amazon Linux compatibility etc.

### ISSUE THREAD

**Issue by jlevy** [open-guides/og-aws#290]
Cover a couple items: CentOS/RedHat version it's based on. Also what about update schedules; do they differ from CentOS/RedHat or Ubuntu etc?

--- 

Saving discussion for later:

bgdnlp [9:31 PM]  
Amazon Linux does NOT use systemd

[9:34]  
Thankfully, as far as I'm concerned. And I don't think they announced any plans for it.

[9:36]  
Amazon Linux is based on centos, centos is not backwards compatible between major versions. Meaning, CentOS 7 (the systemd one) is not compatible with Centos 6, there is no upgrade path from 6 to 7, you have to install a new machine

[9:36]  
same for 5 to 6

[9:38]  
end of life for centos 6 is 2020, so I'm guessing Amazon Linux will stay like this for a while more

[9:40]  
and if they did stealthily switch to systemd in amazon linux, I'm sure the forums would be in uproar in the next 5 minutes. we'd know.

vjain8905 [1:41 PM] 
https://forums.aws.amazon.com/thread.jspa?threadID=51647

[1:41]  
>Peter is partly correct. The Amazon Linux AMI is based on RHEL 5.x and parts of RHEL6. One of our goals is binary compatibility with RHEL 5.x, and therefore CentOS5.x. Astute users will note that our kernel is based on 2.6.34, and we have engineered the image to conform to a cloud environment. For example, the lack of Xorg support helps to keep the images small and lean. The goal of the Amazon Linux AMI is to provide an image for use in the cloud and to serve as reference image of EC2 best practices. The maintenance (security, enhancements, features and bug fixes) for the image will come directly from Amazon, while maintaining maximum compatibility, security and functionality.

[1:41]  
I myself haven't been able to find any official documentation on AWS linux's actual upstream. But that's an AWS employee who posted that

Yeah, saw that… but it’s 6 years old (!).

It also states that in 2013, their goal was to be RHEL6 compatible as possible

---

jlevy [1:47 PM]  
[1:48]  
@vjain8905 came across last comment here, but unsure what’s accurate http://blog.schlomo.schapiro.org/2015/01/comparing-amazon-linux.html (edited)

caseybecking [1:50 PM]  
:slightly_smiling_face:

vjain8905 [1:52 PM]  
> Remember that the Amazon Linux AMI repository structure is configured to deliver a continuous flow of updates that allow you to roll from one version of the Amazon Linux AMI to the next.

[1:53]  
src: https://aws.amazon.com/amazon-linux-ami/2016.09-release-notes/

[1:54]  
I interpret that as the Amazon Linux AMI is meant to be "yum update" without breakages continually. I know immutable infrastructures are the hot-thing nowadays, so a user would update/test locally before building and deploying a new AMI (such as using Packer)

[1:55]  
Also, AWS wants you to look at your instances as Cattle, not pets. You kill [and eat] your cattle, you coddle your pets... so ya, your infrastructure should be destructible + recoverable (and ideally not existing for 6 months :P). I hope this makes sense

[1:55]  
I can't find anything pointing at 6 months patching online (edited)

**Comment by vjain8905** [comment]
Adding the rest of the conversation:


jlevy [1:56 PM]  
Yes, agreed.... but also it is still a factor when thinking things through.
That is, how long can you run “yum update” or “apt-get update” or whatever and expect it to behave reasonably.

vjain8905 [1:57 PM]  
Based off the release-notes and quote. You can launch a 2011 version, and in theory, be able to yum-update till the end of time

jlevy [1:57 PM]  
Pets vs cattle is generally good philosophy but fact is not all servers are cattle everywhere, so it’s something to think about. With Ubuntu for example it’s always good practice for internal systems (say VPN or your Jira server) to be LTS releases.

vjain8905 [1:57 PM]  
I read that as "I launched a 2011 AMI in 2016. I did a yum-update and it became a 2016.09"

[1:58]  
It's a different product lifecycle than Ubuntu

[1:58]  
I'm still wrapping my head around this, as I've almost always used SUSE/Ubuntu/RHEL in my work

[1:59]  
To verify my point... I'll try spinning up a 2011 instance and see what happens

jlevy [2:01 PM]  
Cool. I’m capturing the info here so we con’t lose track (gotta run for a bit). https://github.com/open-guides/og-aws/issues/290

[2:02]  
This should ultimately just be a couple bullet points about Amazon Linux and how it compares with other releases, I think.

pw [2:12 PM]  
I know someone who works in AWS on Amazon Linux, if there are specific questions I can see if he will answer?

vjain8905 [2:13 PM]  
```VERSION: 2011.09
ec2-user@ip-172-31-33-108 ~]$ uname -a
Linux ip-172-31-33-108 2.6.35.14-95.38.amzn1.x86_64 #1 SMP Thu Aug 25 17:11:23 UTC 2011 x86_64 x86_64 x86_64 GNU/Linux

$ sudo yum clean all
$ sudo yum update

https://aws.amazon.com/amazon-linux-ami/2016.09-release-notes/
[ec2-user@ip-172-31-33-108 ~]$ uname -a
Linux ip-172-31-33-108 4.4.23-31.54.amzn1.x86_64 #1 SMP Tue Oct 18 22:02:09 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

[2:13]  
Just did a test, 2011.09 updated to 2016.09 without issue

[2:13]  
Testing 2013.09  now

[2:15]  
 ```VERSION: 2013.09
https://aws.amazon.com/amazon-linux-ami/2013.09-release-notes/
42 package(s) needed for security, out of 210 available
Run "sudo yum update" to apply all updates.
Amazon Linux version 2016.09 is available.

[ec2-user@ip-172-31-24-250 ~]$ uname -a
Linux ip-172-31-24-250 3.4.68-59.97.amzn1.x86_64 #1 SMP Tue Nov 5 07:40:09 UTC 2013 x86_64 x86_64 x86_64 GNU/Linux

$ sudo yum clean all
$ sudo yum update

https://aws.amazon.com/amazon-linux-ami/2016.09-release-notes/
[ec2-user@ip-172-31-24-250 ~]$ uname -a
Linux ip-172-31-24-250 4.4.23-31.54.amzn1.x86_64 #1 SMP Tue Oct 18 22:02:09 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux```

[2:15]  
TLDR; Regardless of the version of Amazon Linux you launch, you CAN simply `yum update` to get to the latest release without major issues.

[2:18]  
As another data-point, doing a "yum install php" installed 5.3.29 and not 5.6, 6.0 (lol) or 7.0

vjain8905 [2:33 PM]  
Bit of a broad stroke answer, but I hope that answers the question in a roundabout way

**Comment by bgdnlp** [comment]
Amazon Linux is based on (started as?) CentOS / RHEL / Fedora 6. I'm pretty sure that it was saying that at some point, but it now probably differs too much from CentOS and branding it as such would cause issues. Either way, the packages that it uses are compatible with CentOS / RHEL / Fedora 6. The EPEL repository that one can enable in /etc/yum.repos.d/epel.repo is for version 6.

The /etc/os-release file identifies it as Amazon Linux, like "rhel fedora":
```
~]$ cat /etc/os-release
NAME="Amazon Linux AMI"
VERSION="2016.09"
ID="amzn"
ID_LIKE="rhel fedora"
VERSION_ID="2016.09"
PRETTY_NAME="Amazon Linux AMI 2016.09"
ANSI_COLOR="0;33"
CPE_NAME="cpe:/o:amazon:linux:2016.09:ga"
HOME_URL="http://aws.amazon.com/amazon-linux-ami/"
```

As for pros and cons:
- It has a number of things installed to make it work better on AWS. Things like kernel modules or drivers written specifically for AWS hardware, applications that make it more friendly to cloud environments in general, tuning of different parameters to improve performance on AWS. Other distributions may or may not have at least some of these applied.
- It is a rolling release, meaning that packages will be upgraded as improvements become available. Snapshots are taken every three months and made available as new AMIs.
- It is only available in AWS, which means that it might cause lock in. It is compatible with RHEL 6 to some extent, so it very much depends on use case.
